python/lsst/afw/table/slots.cc

```diff
@@ -31,48 +30,41 @@ using namespace pybind11::literals;
 namespace lsst {
 namespace afw {
 namespace table {
-
 namespace {
-/**
-Declare standard methods for subclasses of SlotDefinition (but not SlotDefinition itself).
-*/
-template<typename SlotDefSubclass>
-void declareSlotDefinitions(py::class_<SlotDefSubclass, SlotDefinition> & cls) {
-    cls.def("isValid", &SlotDefSubclass::isValid);
-    cls.def("getMeasKey", &SlotDefSubclass::getMeasKey);
-    cls.def("getErrKey", &SlotDefSubclass::getErrKey);
-    cls.def("getFlagKey", &SlotDefSubclass::getFlagKey);
-    cls.def("setKeys", &SlotDefSubclass::setKeys, "alias"_a, "schema"_a);
-}
 
-}  // anonymous namespace
+// tiny classes only used in afw::table: no need for shared_ptr.
+using PySlotDefinition = py::class_<SlotDefinition>;
 
-PYBIND11_PLUGIN(_slots) {
-    py::module mod("_slots", "Python wrapper for afw _slots library");
-
-    /* Module level */
-    py::class_<SlotDefinition> clsSlotDefinition(mod, "SlotDefinition");
-    py::class_<FluxSlotDefinition, SlotDefinition> clsFluxSlotDefinition(mod, "FluxSlotDefinition");
-    py::class_<CentroidSlotDefinition, SlotDefinition>
-        clsCentroidSlotDefinition(mod, "CentroidSlotDefinition");
-    py::class_<ShapeSlotDefinition, SlotDefinition> clsShapeSlotDefinition(mod, "ShapeSlotDefinition");
-
-    /* Member types and enums */
-
-    /* Constructors */
-    clsSlotDefinition.def(py::init<std::string const &>(), "name"_a);
+void declareSlotDefinition(py::module & mod) {
+    PySlotDefinition cls(mod, "SlotDefinition");
+    cls.def("getName", &SlotDefinition::getName);
+    cls.def("getAlias", &SlotDefinition::getAlias);
+}
 
-    /* Operators */
+/*
+Declare standard methods for subclasses of SlotDefinition (but not SlotDefinition itself).
+*/
+template <typename Class>
+void declareSlotDefinitionSubclass(py::module & mod, std::string const & name) {
+    py::class_<Class, SlotDefinition> cls(mod, name.c_str());
+    cls.def(py::init<std::string const &>(), "name"_a);
+    cls.def("isValid", &Class::isValid);
+    cls.def("getMeasKey", &Class::getMeasKey);
+    cls.def("getErrKey", &Class::getErrKey);
+    cls.def("getFlagKey", &Class::getFlagKey);
+    cls.def("setKeys", &Class::setKeys, "alias"_a, "schema"_a);
+}
 
-    /* Members */
-    clsSlotDefinition.def("getName", &SlotDefinition::getName);
-    clsSlotDefinition.def("getAlias", &SlotDefinition::getAlias);
+PYBIND11_PLUGIN(slots) {
```

Comment:
Why is this in anonymous namespace? A code comment would be helpful (here and anywhere else it's done -- I noted one other instance above)